streaming.go

```diff
@@ -10,7 +10,10 @@ import (
 )
 
 type requestStream struct {
-	header          *RequestHeader
+	header interface {
+		ContentLength() int
+		ReadTrailer(r *bufio.Reader) error
+	}
```

Comment:
Probably better to define an interface type for this as it's used twice in this file.

Reply:
Thank you. I have modified it.